### PR TITLE
Adding python wrappers and notebook example for TuneHyperparameters

### PR DESCRIPTION
**Pull request by imatiach-msft**
Adding python wrappers and notebook example for TuneHyperparameters

### CODE REVIEW

File: src/tune-hyperparameters/src/main/python/HyperparamBuilder.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        return self.hyperparamBuilder.build()
+
+class DiscreteHyperParam(object):
+    """
+    Specifies a discrete list of values.
+    """
+    def __init__(self, values, seed=0):
+        ctx = SparkContext.getOrCreate()
+        self.jvm = ctx.getOrCreate()._jvm
+        valuesList = self.jvm.scala.collection.JavaConversions.asScalaBuffer(values).toList()
```
Comment (mhamilton723) [inline src/tune-hyperparameters/src/main/python/HyperparamBuilder.py]: Nit: add ArrayList constructor

File: src/tune-hyperparameters/src/main/python/HyperparamBuilder.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+class RangeHyperParam(object):
+    """
+    Specifies a range of values.
+    """
+    def __init__(self, min, max, seed=0, isDouble=False):
+        ctx = SparkContext.getOrCreate()
+        self.jvm = ctx.getOrCreate()._jvm
+        if (isinstance(min, int)):
+            self.rangeParam = self.jvm.com.microsoft.ml.spark.IntRangeHyperParam(min, max, seed)
```
Comment (mhamilton723) [inline src/tune-hyperparameters/src/main/python/HyperparamBuilder.py]: is there any way to make sure this logic happens on the scala side?

File: src/tune-hyperparameters/src/main/python/HyperparamBuilder.py (hunk 1 above)
Comment (imatiach-msft) [reply src/tune-hyperparameters/src/main/python/HyperparamBuilder.py]: done

File: src/tune-hyperparameters/src/main/python/HyperparamBuilder.py (hunk 2 above)
Comment (imatiach-msft) [reply src/tune-hyperparameters/src/main/python/HyperparamBuilder.py]: done!

File: src/tune-hyperparameters/src/main/scala/HyperparamBuilder.scala [hunk 3]
```diff
@@ -54,6 +54,36 @@ class DoubleRangeHyperParam(min: Double, max: Double, seed: Long = 0)
...
+object HyperParamUtils {
+  /**
+    * Returns a range hyper param by matching to the given input type.
+    * @param min The min value of the range
+    * @param max The max value of the range
+    * @param seed The random number seed.
+    * @return A RangeHyperParam matched to the given type for min and max values.
+    */
+  def getRangeHyperParam(min: Any, max: Any, seed: Long = 0): RangeHyperParam[_] = {
+    min match {
```
Comment (mhamilton723) [inline src/tune-hyperparameters/src/main/scala/HyperparamBuilder.scala]: You can use scala pattern matching to your advantage here: try something like

(min, max) match {
    case (a: Float, b: Float) => floatRange(a,b)
   ......

**Comment by mhamilton723** [comment]
Very Close, just one Nit then feel free to merge

File: src/tune-hyperparameters/src/main/scala/HyperparamBuilder.scala (hunk 3 above)
Comment (imatiach-msft) [reply src/tune-hyperparameters/src/main/scala/HyperparamBuilder.scala]: good idea, done!

**Pull request merged**